mcs/class/System/Test/System.Diagnostics/PerformanceCounterTests.cs

```diff
@@ -0,0 +1,148 @@
+{
+	internal class Common
+	{
+		internal string guid;
+		internal string baseName;
+		internal string name;
+		internal string category;
+		internal CounterCreationDataCollection counterDataCollection;
+		internal CounterCreationData counter;
+		internal CounterCreationData counterBase;
+		internal PerformanceCounter counterSample;
+		internal PerformanceCounterCategory pcc;
+
+		internal void DoCommon()
+		{
+			guid = Guid.NewGuid().ToString("N");
+			name = guid + "_Counter";
+			category = name + "_Category";
+			counterDataCollection = new CounterCreationDataCollection();
+
+			// Add the counter.
+			counter = new CounterCreationData();
+			counter.CounterType = PerformanceCounterType.AverageCount64;
+			counter.CounterName = name;
+			counterDataCollection.Add(counter);
+
+			// Add the base counter.
+			counterBase = new CounterCreationData();
+			counterBase.CounterType = PerformanceCounterType.AverageBase;
+			baseName = name + "Base";
+			counterBase.CounterName = baseName;
+			counterDataCollection.Add(counterBase);
+
+			// Create the category.
+			PerformanceCounterCategory.Create(category,
+				"description",
+				PerformanceCounterCategoryType.SingleInstance, counterDataCollection);
+
+			counterSample = new PerformanceCounter(category,
+				name, false);
+
+			counterSample.RawValue = 0;
+
+			pcc = new PerformanceCounterCategory(category);
+		}
+	}
+
+	[TestFixture]
+	public static class PerformanceCounterTests
+	{
+		[Test]
+		public static void PerformanceCounterCategory_CreateCategory()
+		{
+			string name = "AverageCounter64Sample";
+
+			Assert.False(PerformanceCounterCategory.Exists(name + "Category"));
+
+			CounterCreationDataCollection counterDataCollection = new CounterCreationDataCollection();
+
+			// Add the counter.
+			CounterCreationData averageCount64 = new CounterCreationData();
+			averageCount64.CounterType = PerformanceCounterType.AverageCount64;
+			averageCount64.CounterName = name;
+			counterDataCollection.Add(averageCount64);
+
+			// Add the base counter.
+			CounterCreationData averageCount64Base = new CounterCreationData();
+			averageCount64Base.CounterType = PerformanceCounterType.AverageBase;
+			averageCount64Base.CounterName = name + "Base";
+			counterDataCollection.Add(averageCount64Base);
+
+			// Create the category.
+			PerformanceCounterCategory.Create(name + "Category",
+			"Demonstrates usage of the AverageCounter64 performance counter type.",
+			PerformanceCounterCategoryType.SingleInstance, counterDataCollection);
+
+			Assert.True(PerformanceCounterCategory.Exists(name + "Category"));
+		}
+
+		[Test]
+		public static void PerformanceCounter_CreateCounter_Count0()
+		{
+			var a = new Common();
+			a.DoCommon();
+			Assert.AreEqual(0, a.counterSample.RawValue);
+			a.counterSample.Increment();
+			Assert.AreEqual(1, a.counterSample.RawValue);
+		}
+
+		public static bool verbose = Environment.GetEnvironmentVariable("MONO_TEST_VERBOSE") != null;
```

Comment:
please use the `V` environment variable instead, we use this for enabling verbose output in more places.